accelerator/apis.c

```diff
@@ -214,6 +214,91 @@ status_t api_receive_mam_message(const iota_client_service_t* const service,
   return ret;
 }
 
+status_t api_mam_send_message(const iota_config_t* const tangle,
+                              const iota_client_service_t* const service,
+                              char const* const payload, const bool last_packet,
+                              tryte_t** bundle_hash_result,
+                              tryte_t** channel_id_result) {
+  status_t ret = SC_OK;
+  mam_api_t mam;
+  bundle_transactions_t* bundle = NULL;
+  mam_psk_t_set_t psks = NULL;
+  bundle_transactions_new(&bundle);
+  tryte_t channel_id[MAM_CHANNEL_ID_SIZE];
+  trit_t msg_id[MAM_MSG_ID_SIZE];
+  tryte_t* payload_trytes =
+      (tryte_t*)malloc(2 * strlen(payload) * sizeof(tryte_t));
+  if (!payload_trytes) {
+    return SC_MAM_OOM;
+  }
+  ascii_to_trytes(payload, payload_trytes);
+  size_t payload_size = strlen(payload) * 2;
```

Comment:
You should avoid computing the length of `payload` repeatedly. Also, be aware of security risks.

Reply:
I wonder what kind of security issue would happen here?

Reply:
typical buffer overflow. Further reading:
* [CVE-2017-1000257](https://nvd.nist.gov/vuln/detail/CVE-2017-1000257)
* [CVE-2017-9047](https://www.cvedetails.com/cve/CVE-2017-9047/)